handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslContext.java

```diff
@@ -637,29 +637,36 @@ static void setKeyMaterial(long ctx, X509Certificate[] keyCertChain, PrivateKey
          /* Load the certificate file and private key. */
         long keyBio = 0;
         long keyCertChainBio = 0;
+        long keyCertChainBio2 = 0;
 
         try {
             keyCertChainBio = toBIO(keyCertChain);
+            keyCertChainBio2 = toBIO(keyCertChain);
+
             keyBio = toBIO(key);
 
             SSLContext.setCertificateBio(
                     ctx, keyCertChainBio, keyBio,
                     keyPassword == null ? StringUtil.EMPTY_STRING : keyPassword);
             // We may have more then one cert in the chain so add all of them now.
-            SSLContext.setCertificateChainBio(ctx, keyCertChainBio, false);
+            SSLContext.setCertificateChainBio(ctx, keyCertChainBio2, false);
         } catch (SSLException e) {
             throw e;
         } catch (Exception e) {
             throw new SSLException("failed to set certificate and key", e);
         } finally {
-            if (keyBio != 0) {
-                SSL.freeBIO(keyBio);
-            }
-            if (keyCertChainBio != 0) {
-                SSL.freeBIO(keyCertChainBio);
+            freeBio(keyBio, keyCertChainBio, keyCertChainBio2);
+        }
+    }
+
+    static void freeBio(long... bios) {
```

Comment:
@Scottmitch actually I was wrong this will not be optimised away. Maybe let me just call it x times by hand.